Runs: Rewrite Runs section (#2011)
wandb/docs: models/runs/stop-runs.mdx

---
title: Stop runs
description: Learn how to stop runs programmatically using the W&B Python SDK or manually from the W&B App.
---

> Stop a run programmatically with the W&B Python SDK or manually from the W&B App.

Reviewer: ```suggestion
Stop a run programmatically with the W&B Python SDK or interactively in the W&B App.
```

<Tabs>
<Tab title="Python SDK">

> 1. Navigate to the terminal or code editor where you initialized the run.
> 2. Press `Ctrl+D` to stop the run.

Reviewer: ```suggestion
In the terminal or code editor where you initialized the run, press `Ctrl+D` to stop the run.
```

For example, following the previous instructions, your terminal might looks similar to the following: 

```bash
KeyboardInterrupt
wandb: 🚀 View run legendary-meadow-2 at: https://wandb.ai/nico/history-blaster-4/runs/o8sdbztv
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 1 other file(s)
wandb: Find logs at: ./wandb/run-20241106_095857-o8sdbztv/logs
```

> Navigate to the W&B App to confirm the run is no longer active:

Reviewer: This is weird. So you actually need to use the App whether you are doing it programmatically or interactively? Seems a little silly. If the run is no longer active, wouldn't the terminal have returned already?

Author: Not necessarily. Sweeps (for e.g.) can seem finished, but they're not. There are other times, too when a run "hangs", e.g. if you don't use a context manager and don't explicitly call run.finish() in a notebook.

1. Navigate to the project that your run was logging to.
2. Select the name of the run. 

>   <Note>

Reviewer: Again, this is weird. If you are in the app we are directing you to the terminal, and above in the terminal instructions we tell you to check the app first.

---
title: Stop runs
description: Stop runs programmatically using the W&B Python SDK or manually from the W&B App.
---

Stop a run programmatically with the W&B Python SDK or interactively in the W&B App.

> <Note tip>

Reviewer: ?
